### ISSUE TITLE
Drupal fails to install

### ISSUE THREAD

**Issue by magick93** [jubicoy/drupal8-docker#5]
### Versions

* **OpenShift Master**: v1.5.0-alpha.3+cf7e336
* **Kubernetes Master** v1.5.2+43a9be4

When we try to run this template, D8 doesnt install. Instead we see the following errors in the web container:

```
Running drush updb
Error: Call to undefined function drupal_get_installed_schema_version() in /var/www/drupal/core/includes/install.inc on line 80 #0 phar:///usr/local/bin/drush/commands/core/drupal/update.inc(130): drupal_load_updates()
#1 phar:///usr/local/bin/drush/commands/core/core.drush.inc(462): update_main()
#2 phar:///usr/local/bin/drush/includes/command.inc(422): drush_core_updatedb()
#3 phar:///usr/local/bin/drush/includes/command.inc(231): _drush_invoke_hooks(Array, Array)
#4 phar:///usr/local/bin/drush/includes/command.inc(199): drush_command()
#5 phar:///usr/local/bin/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch(Array)
#6 phar:///usr/local/bin/drush/includes/preflight.inc(66): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#7 phar:///usr/local/bin/drush/includes/startup.inc(458): drush_main()
#8 phar:///usr/local/bin/drush/includes/startup.inc(365): drush_run_main(false, '/', 'Phar detected. ...')
#9 phar:///usr/local/bin/drush/drush(114): drush_startup(Array)
#10 /usr/local/bin/drush(10): require('phar:///usr/loc...')
#11 {main}
Error: Call to undefined function drupal_get_installed_schema_version() in drupal_load_updates() (line 80 of /var/www/drupal/core/includes/install.inc).
Drush command terminated abnormally due to an unrecoverable error.       [error]
Running drush entup
Command entity-updates needs a higher bootstrap level to run - you       [error]
will need to invoke drush from a more functional Drupal environment
to run this command.
The drush command 'entup' could not be executed.                         [error]
Drush was not able to start (bootstrap) the Drupal database.             [error]
Hint: This may occur when Drush is trying to:
 * bootstrap a site that has not been installed or does not have a
configured database. In this case you can select another site with a
working database setup by specifying the URI to use with the --uri
parameter on the command line. See `drush topic docs-aliases` for
details.
 * connect the database through a socket. The socket file may be
wrong or the php-cli may have no access to it in a jailed shell. See
http://drupal.org/node/1428638 for details.
Drush was attempting to connect to: 
 Drupal version         :  8.2.7                      
 Site URI               :  http://default             
 PHP configuration      :  /etc/php/7.0/cli/php.ini   
 PHP OS                 :  Linux                      
 Drush script           :  /usr/local/bin/drush       
 Drush version          :  8.1.10                     
 Drush temp directory   :  /tmp                       
 Drush configuration    :                             
 Drush alias files      :                             
 Drupal root            :  /var/www/drupal            
 Drupal Settings File   :  sites/default/settings.php 
 Site path              :  sites/default              
Running drush cr
RuntimeException: Missing $settings['hash_salt'] in settings.php. in     [error]
/var/www/drupal/core/lib/Drupal/Core/Site/Settings.php:140
Stack trace:
#0
/var/www/drupal/core/lib/Drupal/Core/PhpStorage/PhpStorageFactory.php(43):
Drupal\Core\Site\Settings::getHashSalt()
#1 /var/www/drupal/core/includes/utility.inc(32):
Drupal\Core\PhpStorage\PhpStorageFactory::get('twig')
#2 phar:///usr/local/bin/drush/commands/core/cache.drush.inc(300):
drupal_rebuild(Object(Composer\Autoload\ClassLoader),
Object(Symfony\Component\HttpFoundation\Request))
#3 phar:///usr/local/bin/drush/includes/command.inc(422):
drush_cache_rebuild()
#4 phar:///usr/local/bin/drush/includes/command.inc(231):
_drush_invoke_hooks(Array, Array)
#5 phar:///usr/local/bin/drush/includes/command.inc(199):
drush_command()
#6 phar:///usr/local/bin/drush/lib/Drush/Boot/BaseBoot.php(67):
drush_dispatch(Array)
#7 phar:///usr/local/bin/drush/includes/preflight.inc(66):
Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#8 phar:///usr/local/bin/drush/includes/startup.inc(458):
drush_main()
#9 phar:///usr/local/bin/drush/includes/startup.inc(365):
drush_run_main(false, '/', 'Phar detected. ...')
#10 phar:///usr/local/bin/drush/drush(114): drush_startup(Array)
#11 /usr/local/bin/drush(10): require('phar:///usr/loc...')
#12 {main}
Adding password for user priceinsighter
2017-03-22 15:51:50,561 CRIT Set uid to user 104
2017-03-22 15:51:50,561 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2017-03-22 15:51:50,570 INFO RPC interface 'supervisor' initialized
2017-03-22 15:51:50,570 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2017-03-22 15:51:50,571 INFO supervisord started with pid 1
2017-03-22 15:51:51,573 INFO spawned: 'nginx' with pid 58
2017-03-22 15:51:51,574 INFO spawned: 'php7.0-fpm' with pid 59
2017/03/22 15:51:51 [warn] 58#58: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/22 15:51:51 [emerg] 58#58: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-22 15:51:51,590 INFO exited: nginx (exit status 1; not expected)
[22-Mar-2017 15:51:51] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[22-Mar-2017 15:51:51] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
2017-03-22 15:51:52,626 INFO spawned: 'nginx' with pid 62
2017-03-22 15:51:52,626 INFO success: php7.0-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017/03/22 15:51:52 [warn] 62#62: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/22 15:51:52 [emerg] 62#62: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-22 15:51:52,639 INFO exited: nginx (exit status 1; not expected)
2017-03-22 15:51:54,643 INFO spawned: 'nginx' with pid 70
2017/03/22 15:51:54 [warn] 70#70: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/22 15:51:54 [emerg] 70#70: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-22 15:51:54,656 INFO exited: nginx (exit status 1; not expected)
2017-03-22 15:51:57,660 INFO spawned: 'nginx' with pid 71
2017/03/22 15:51:57 [warn] 71#71: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/22 15:51:57 [emerg] 71#71: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-22 15:51:57,673 INFO exited: nginx (exit status 1; not expected)
2017-03-22 15:51:58,675 INFO gave up: nginx entered FATAL state, too many start retries too quickly
```

**Comment by jesseik** [comment]
Hi! I get the same error when I try to restart already installed Drupal pod without Mysql pod running.
If you are trying to install a new instance of Drupal, clear the volume and then restart Mysql and Drupal-web pods.

**Comment by magick93** [comment]
I have done the following:

- Deleted contents of volume
- Restarted web pod
- Restarted db pod

The logs for the web container are:


```
cp: cannot stat '/tmp/sites/*': No such file or directory
cp: cannot create regular file '/var/www/drupal/sites/default/settings.local.php': No such file or directory
Copying development.services.yml
Downloading module ctools
Project ctools (8.x-3.0-alpha27) downloaded to                         [success]
/volume/modules/ctools.
Project ctools contains 3 modules: ctools_views, ctools_block, ctools.
Downloading module pathauto
Project pathauto (8.x-1.0-rc1) downloaded to                           [success]
/volume/modules/pathauto.
Downloading module token
Project token (8.x-1.0-rc1) downloaded to /volume/modules/token.       [success]
Downloading module imce
Project imce (8.x-1.5) downloaded to /volume/modules/imce.             [success]
Downloading module honeypot
Project honeypot (8.x-1.23) downloaded to /volume/modules/honeypot.    [success]
Downloading module yamlform
Project yamlform (8.x-1.0-beta29) downloaded to                        [success]
/volume/modules/yamlform.
Project yamlform contains 7 modules: yamlform_node, yamlform_examples, yamlform_ui, yamlform_devel, yamlform_to_webform, yamlform_templates, yamlform.
Downloading module mailsystem
Project mailsystem (8.x-4.1) downloaded to                             [success]
/volume/modules/mailsystem.
Downloading module smtp
Project smtp (8.x-1.0-beta2) downloaded to /volume/modules/smtp.       [success]
Downloading module paragraphs
Project paragraphs (8.x-1.1) downloaded to                             [success]
/volume/modules/paragraphs.
Project paragraphs contains 3 modules: paragraphs_demo, paragraphs_type_permissions, paragraphs.
Downloading module entity
Project entity (8.x-1.0-alpha4) downloaded to /volume/modules/entity.  [success]
Downloading module entity_reference_revisions
Project entity_reference_revisions (8.x-1.2) downloaded to             [success]
/volume/modules/entity_reference_revisions.
Downloading module xmlsitemap
Project xmlsitemap (8.x-1.0-alpha2) downloaded to                      [success]
/volume/modules/xmlsitemap.
Project xmlsitemap contains 3 modules: xmlsitemap_custom, xmlsitemap_engines, xmlsitemap.
Downloading module google_analytics
Project google_analytics (8.x-2.1) downloaded to                       [success]
/volume/modules/google_analytics.
Downloading module metatag
Project metatag (8.x-1.0) downloaded to /volume/modules/metatag.       [success]
Project metatag contains 14 modules: metatag_app_links, metatag_open_graph, metatag_facebook, metatag_dc, metatag_google_plus, metatag_open_graph_products, metatag_twitter_cards, metatag_dc_advanced, metatag_favicons, metatag_mobile, metatag_hreflang, metatag_verification, metatag_google_cse, metatag.
Downloading module background_image_formatter
Project background_image_formatter (8.x-1.1) downloaded to             [success]
/volume/modules/background_image_formatter.
Downloading module advagg
Project advagg (8.x-2.3) downloaded to /volume/modules/advagg.         [success]
Project advagg contains 8 modules: advagg_ext_minify, advagg_validator, advagg_cdn, advagg_mod, advagg_css_minify, advagg_js_minify, advagg_bundler, advagg.
Downloading module menu_block
Project menu_block (8.x-1.4) downloaded to                             [success]
/volume/modules/menu_block.
Downloading module redirect
Project redirect (8.x-1.0-alpha4) downloaded to                        [success]
/volume/modules/redirect.
Project redirect contains 3 modules: redirect_domain, redirect_404, redirect.
Downloading module rules
Project rules (8.x-3.0-alpha2) downloaded to /volume/modules/rules.    [success]
Downloading module google_tag
Project google_tag (8.x-1.x-dev) downloaded to                         [success]
/volume/modules/google_tag.
Downloading module fontyourface
Project fontyourface (8.x-3.1) downloaded to                           [success]
/volume/modules/fontyourface.
Project fontyourface contains 7 modules: fontsquirrel_api, fontscom_api, local_fonts, adobe_edge_fonts, typekit_api, google_fonts_api, fontyourface.
Downloading module hotjar
Project hotjar (8.x-1.0) downloaded to /volume/modules/hotjar.         [success]
Downloading module colorbox
Project colorbox (8.x-1.2) downloaded to /volume/modules/colorbox.     [success]
Downloading module bootstrap
Project bootstrap (8.x-3.2) downloaded to /volume/modules/bootstrap.   [success]
Downloading module adminimal_theme
Project adminimal_theme (8.x-1.3) downloaded to                        [success]
/volume/modules/adminimal_theme.
Downloading theme bootstrap
Project bootstrap (8.x-3.2) downloaded to /volume/themes/bootstrap.    [success]
Downloading theme adminimal_theme
Project adminimal_theme (8.x-1.3) downloaded to                        [success]
/volume/themes/adminimal_theme.
Adding password for user priceinsighter
2017-03-23 09:01:53,373 CRIT Set uid to user 104
2017-03-23 09:01:53,373 WARN Included extra file "/etc/supervisor/conf.d/supervisord.conf" during parsing
2017-03-23 09:01:53,382 INFO RPC interface 'supervisor' initialized
2017-03-23 09:01:53,383 CRIT Server 'unix_http_server' running without any HTTP authentication checking
2017-03-23 09:01:53,383 INFO supervisord started with pid 1
2017-03-23 09:01:54,475 INFO spawned: 'nginx' with pid 690
2017-03-23 09:01:54,477 INFO spawned: 'php7.0-fpm' with pid 691
2017/03/23 09:01:54 [warn] 690#690: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/23 09:01:54 [emerg] 690#690: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-23 09:01:54,491 INFO exited: nginx (exit status 1; not expected)
[23-Mar-2017 09:01:54] NOTICE: [pool www] 'user' directive is ignored when FPM is not running as root
[23-Mar-2017 09:01:54] NOTICE: [pool www] 'group' directive is ignored when FPM is not running as root
2017-03-23 09:01:55,521 INFO spawned: 'nginx' with pid 694
2017-03-23 09:01:55,521 INFO success: php7.0-fpm entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
2017/03/23 09:01:55 [warn] 694#694: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/23 09:01:55 [emerg] 694#694: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-23 09:01:55,534 INFO exited: nginx (exit status 1; not expected)
2017-03-23 09:01:57,537 INFO spawned: 'nginx' with pid 695
2017/03/23 09:01:57 [warn] 695#695: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/23 09:01:57 [emerg] 695#695: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-23 09:01:57,550 INFO exited: nginx (exit status 1; not expected)
2017-03-23 09:02:00,555 INFO spawned: 'nginx' with pid 696
2017/03/23 09:02:00 [warn] 696#696: duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
nginx: [warn] duplicate MIME type "text/html" in /etc/nginx/nginx.conf:30
2017/03/23 09:02:00 [emerg] 696#696: mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
nginx: [emerg] mkdir() "/var/lib/nginx/body" failed (13: Permission denied)
2017-03-23 09:02:00,569 INFO exited: nginx (exit status 1; not expected)
2017-03-23 09:02:01,570 INFO gave up: nginx entered FATAL state, too many start retries too quickly
```

**Comment by jesseik** [comment]
Hi! This should be now fixed with commit 88fffdec87f7db4456fffbf9605a07659c2bab46

**Comment by jesseik** [comment]
@magick93 Is this resolved with commit 88fffdec87f7db4456fffbf9605a07659c2bab46?

**Comment by magick93** [comment]
Yes, thanks
